### PR TITLE
Document special files

### PR DESCRIPTION
**Pull request by ozh**
Will close https://github.com/YOURLS/YOURLS/issues/3045

### CODE REVIEW

**Comment by ozh** [comment]
Feedback welcome on wording and stuff

**Review by LeoColomb — DISMISSED** [review]

File: docs/guide/extend/special-files.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Special files
+
+YOURLS support a number of _special user files_ to further customize some internal behaviors.
+They are all loaded from `user/` folder.
```
Comment (dgw) [inline docs/guide/extend/special-files.md:4]:
```suggestion
They are all loaded from the `user/` folder.
```

File: docs/guide/extend/special-files.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Pages
+
+Pages are custom files is `user/pages/` that will be displayed when requested like short URLs.
```
Comment (dgw) [inline docs/guide/extend/special-files.md:8]:
```suggestion
Pages are custom files in `user/pages/` that will be displayed when requested like short URLs.
```

File: docs/guide/extend/special-files.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+By default YOURLS displays a tiny YOURLS favicon.
+You can easily display a custom favicon with your own `user/favicon.(jpg|gif|png|ico|svg)`.
+
+## Custom "always-on plugin"
+
+If present, YOURLS will always include `user/cache.php` early during the boot process, before regular active plugins are loaded.
+
+This special file will not appear in the list of plugin and cannot be activated or deactivated in the admin area. To deactivate it,
+delete (or rename) it manually.
```
Comment (dgw) [inline docs/guide/extend/special-files.md:31]:
```suggestion
This special file will not appear in the list of plugins and cannot be activated or deactivated in the admin area.
To deactivate it, delete (or rename) it manually.
```

File: docs/guide/extend/special-files.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+:::note
+"Cache" stands for "Custom Additional Code for Hazardous Extension".
```
Comment (dgw) [inline docs/guide/extend/special-files.md:34]:
```suggestion
"Cache" stands for "Custom Additional Code for Hazardous Extensions".
```

File: docs/guide/extend/special-files.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+:::
+
+## Custom DB layer
+
+Out of the box, YOURLS uses PDO to interact with a MySQL server.
+If you are an experienced coder and want to replace the core DB engine, you can implement your own library in `/user/db.php`.
```
Comment (dgw) [inline docs/guide/extend/special-files.md:40]:
```suggestion
If you are an experienced coder and want to replace the core DB engine, you can implement your own library in `user/db.php`.
```

None of the other examples have the leading slash. 🤷‍♂️

**Review by dgw — COMMENTED** [review]
Tweaks & typo fixes

**Review by dgw — APPROVED** [review]

**Review by LeoColomb — APPROVED** [review]

**Comment by ozh** [comment]
👍

**Pull request merged**